Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1811,6 +1778,80 @@ void CGContextEOFillPath(CGContextRef context) {
 }
 #pragma endregion
 
+#pragma region Drawing Operations - CGImage
+
+/**
+ @Status Interoperable
+*/
+void CGContextDrawImage(CGContextRef context, CGRect rect, CGImageRef image) {
+    NOISY_RETURN_IF_NULL(context);
+    RETURN_IF(!image);
+
+    // Obtain the pixel format for the image
+    WICPixelFormatGUID imagePixelFormat = _CGImageGetPixelFormat(image);
+
+    CFRetain(image);
+    woc::unique_cf<CGImageRef> refImage;
+    refImage.reset(image);
+
+    if (!_CGIsValidRenderTargetPixelFormat(imagePixelFormat)) {
+        // convert it to a valid pixelformat
+        refImage.reset(_CGImageCreateCopyWithPixelFormat(image, GUID_WICPixelFormat32bppPBGRA));
+    }
+
+    ComPtr<IWICBitmap> bmap = _CGImageGetImageSource(refImage.get());
+    ComPtr<ID2D1Bitmap> d2dBitmap;
+
+    HRESULT result = context->RenderTarget()->CreateBitmapFromWicBitmap(bmap.Get(), nullptr, &d2dBitmap);
+
+    CGAffineTransform flipImage = CGAffineTransformMakeTranslation(rect.origin.x, rect.origin.y + (rect.size.height / 2.0));
+    flipImage = CGAffineTransformScale(flipImage, 1.0, -1.0);
+    flipImage = CGAffineTransformTranslate(flipImage, -rect.origin.x, -(rect.origin.y + (rect.size.height / 2.0)));
+
+    ComPtr<ID2D1CommandList> commandList;
+    HRESULT hr = __CGContextRenderToCommandList(context,
```

Comment:
oh god, yes. I removed this accidently for some testing forgot to revert

---
In reply to: [88733470](https://github.com/Microsoft/WinObjC/pull/1414#discussion_r88733470) [](ancestors = 88733470)